experimental/oak_async/tests/fake_runtime.rs

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//! Provides implementations of Oak ABI functions to test the async executor against.
```

Comment:
How hard would it be to convert the tests to run on the real Runtime and not need this?

Reply:
This might be doable, though I don't know how easy it would be to insert errors into certain ABI calls.

I think that even if it would be possible, having a fake runtime is still valuable:
- tests are likely to be simpler to write and maintain
- tests are much to run, thus help with the edit-compile-run cycle
- helps track down bugs in the full system. i.e. if some async code works on the fake runtime but not in a real application, it is likely that the bug is in the runtime, not the executor.